kernel/src/lib.rs

```diff
@@ -90,6 +90,7 @@ pub mod checkpoint;
 pub mod engine_data;
 pub mod error;
 pub mod expressions;
+pub mod log_compaction;
```

Comment:
nit: i know we have gotten in the pattern of lots of 'pub mods' but I think generally I will try to advocate for private modules and just re-export the functionality we need. So here we could just do `mod log_compaction` and just pull in the narrow piece we need in `snapshot.rs`